funcx_endpoint/funcx_endpoint/endpoint/endpoint.py

```diff
@@ -416,18 +416,53 @@ def stop_endpoint(
             exit(-1)
 
     @staticmethod
-    def delete_endpoint(endpoint_dir: pathlib.Path):
+    def delete_endpoint(
+        endpoint_dir: pathlib.Path, endpoint_config: Config | None, force: bool = False
+    ):
         ep_name = endpoint_dir.name
 
         if not endpoint_dir.exists():
             log.warning(f"Endpoint <{ep_name}> does not exist")
             exit(-1)
 
-        # stopping the endpoint should handle all of the process cleanup before
-        # deletion of the directory
+        endpoint_id = Endpoint.get_endpoint_id(endpoint_dir)
+        if endpoint_id is None:
+            log.warning(f"Endpoint <{ep_name}> could not be located")
+            if not force:
+                exit(-1)
+
+        # Delete endpoint from web service
+        try:
+            fx_client = Endpoint.get_funcx_client(endpoint_config)
+            fx_client.delete_endpoint(endpoint_id)
+        except GlobusAPIError as e:
+            log.warning(
+                f"Endpoint <{ep_name}> could not be deleted from web service."
+                + f"  [{e.message}]"
+            )
+            if not force:
+                exit(os.EX_UNAVAILABLE)
+        except NetworkError as e:
+            log.warning(
+                f"Endpoint <{ep_name}> could not be deleted from web service."
+                + f"  [{e}]"
+            )
+            if not force:
+                log.critical(
+                    "funcx-endpoint is unable to reach the funcX service due to a "
```

Comment:
Hmm.  Why manually break the lines?  For clarity?  I'm not sure I have an opinion but I don't think we usually do this?

Reply:
This was actually copied from the [start_endpoint](https://github.com/funcx-faas/funcX/blob/57e49b782543c74b4c50ffb6de2a9f8fc77ce69b/funcx_endpoint/funcx_endpoint/endpoint/endpoint.py#L264-L271) function. I think the line breaks are OK, so I'll leave it as-is.